### DISCUSSION TITLE
Using f-strings in doc strings in FastAPI endpoints does not render docstring in openapi.json

### DISCUSSION THREAD

**Discussion by misspenalty** [fastapi/fastapi#10990] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi

app = fastapi.FastAPI()

TEST_STRING = "my constant"

@app.get("/")
async def root():
    f"""Test doc string for my root endpoint, {TEST_STRING}"""
    return {"message": "Hello World"}
```

### Description

If you use an f-string in the docstring of a fastapi endpoint, the docstring is not rendered at all in the `openapi.json` file.

If you don't use an f-string, the doc string is correctly rendered in the "description" field of the endpoint. Does somebody know if there is a way to get f-strings to be correctly rendered as the "description" of the API endpoint in the `openapi.json` file or if this is maybe a bug in rendering the documentation? Thank you!

My expected output:
```
{
  "openapi": "3.1.0",
  "info": {
    "title": "FastAPI",
    "version": "0.1.0"
  },
  "paths": {
    "/": {
      "get": {
        "summary": "Root",
        "description": "Test doc string for my root endpoint, my constant",
        "operationId": "root__get",
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {}
              }
            }
          }
        }
      }
    }
  }
}
```

The actual output:
```
{
  "openapi": "3.1.0",
  "info": {
    "title": "FastAPI",
    "version": "0.1.0"
  },
  "paths": {
    "/": {
      "get": {
        "summary": "Root",
        "operationId": "root__get",
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {}
              }
            }
          }
        }
      }
    }
  }
}
```

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

Python 3.10.6

**Comment by jgould22** [comment, marked as answer]
f-strings are _not_ doc strings so they do not get added to the `__doc__` attribute by python which is where the data for the openapi schema is probably read.

This is not a bug.

You may be able to populate the `__doc__` attribute some other way which might get you what you want though.
